daemon/containerd/resolver.go

```diff
@@ -33,12 +41,7 @@ func hostsWrapper(hostsFn docker.RegistryHosts, authConfig *registrytypes.AuthCo
 
 		for i := range hosts {
 			if hosts[i].Authorizer == nil {
-				var opts []docker.AuthorizerOpt
-				if authConfig != nil {
-					opts = append(opts, authorizationCredsFromAuthConfig(*authConfig))
-				}
-				hosts[i].Authorizer = docker.NewDockerAuthorizer(opts...)
-
+				hosts[i].Authorizer = authorizer
```

Comment:
Yeah, I was also thinking about that. On the other hand, since this function call is done in containerd, so we don't fully control when it's called, it felt appropriate to be a little bit more defensive than to potentially leak the credentials.

Reply:
Yeah, so my thinking was along these lines (and we could still have the check in the Authoriser itself)

```go
cfgHost := registry.ConvertToHostname(*authConfig.ServerAddress)
if cfgHost == "" || cfgHost == registry.IndexHostname {
	cfgHost = registry.DefaultRegistryHost
}
if cfgHost == host {
	hosts[i].Authorizer = authorizer
}
```

Either way, not super important, but was looking at the code, and thinking.. we're always passing the authorizer, even if we would already know it would never match.

Something to look into In a follow-up (also not 100% sure if "mirrors" could be in effect here)